templates/default/my.cnf.erb

```diff
@@ -19,6 +19,7 @@
 [client]
 port		= <%= node['mariadb']['client']['port']   %>
 socket		= <%= node['mariadb']['client']['socket'] %>
+host		= <%= node['mariadb']['client']['host'] ||= 'localhost' %>
```

Comment:
The client's default is localhost by the way. We can choose to not write it in the config file if the attribute is nil.